tests/Integration/Framework/Test_ProxyManager.py

```diff
@@ -0,0 +1,340 @@
+        if not retVal['OK']:
+          gLogger.warn(retVal['Message'])
+          if 'bad decrypt' in retVal['Message']:
+            return S_ERROR("Bad passphrase")
+          return S_ERROR("Can't load %s" % userKeyFile)
+        return chain.generateProxyToFile(self.proxyPath, time * 3600,
+                                         limited=limit, diracGroup=group,
+                                         rfc=rfc)
+      else:
+        cmd = 'voms-proxy-fake --cert %s --key %s -q' % (userCertFile, userKeyFile)
+        cmd += ' -hostcert %s -hostkey %s' % (self.hostCert, self.hostKey)
+        cmd += ' -uri fakeserver.cern.ch:15000'
+        cmd += ' -fqan "/%s/Role=user/Capability=NULL" -voms %s ' % (vo, vo)
+        cmd += ' -hours %s -out %s' % (time, self.proxyPath)
+        if limit:
+          cmd += ' -limited'
+        if rfc:
+          cmd += ' -rfc'
+        status, output = commands.getstatusoutput(cmd)
+        if status:
+          return S_ERROR(output)
+        return S_OK()
+
+    # Create certificates
+    for user in ['no_user', 'user_11', 'user_21']:
+      result = createCert(user)
+      self.assertTrue(result['OK'], 'Cannot create certificate for "%s": %s' % (user, result.get('Message')))
+    
+    userUpload = 'user_11'
+    userProxy = 'user_11'
+    groupUpload = 'dirac_user_1'
+    groupProxy = False
+    rfcUpload = True
+    rfcProxy = True
+    limitUpload = False
+    limitProxy = False
+    timeUpload = 12
+    timeProxy = 12
+    voUpload = 'vo_1'
+    voProxy = False
+
+    timeReq = 12 * 3600
+    rfcReq = True
+
+
+    result = createProxyToFile(userUpload, groupUpload, rfcUpload, limitUpload, timeUpload, voUpload, '/tmp/x509up_u%s' % os.geteuid())
+    uploadParams = 'User: %s\nGroup: %s\nRFC: %s\nLimited: %s\nLivetime: %s\nVO: %s' % (userUpload, groupUpload, rfcUpload, limitUpload, timeUpload, voUpload)
+    self.assertTrue(result['OK'], 'Cannot create proxy with next params:\n %s\nERROR: %s' % (uploadParams, result.get('Message')))
+    
+    result = createProxyToFile(userProxy, groupProxy, rfcProxy, limitProxy, timeProxy, voProxy)
+    proxyParams = 'User: %s\nGroup: %s\nRFC: %s\nLimited: %s\nLivetime: %s\nVO: %s' % (userProxy, groupProxy, rfcProxy, limitProxy, timeProxy, voProxy)
+    self.assertTrue(result['OK'], 'Cannot create proxy that need to attach with next params:\n %s\nERROR: %s' % (proxyParams, result.get('Message')))
+    
+    chain = X509Chain()
+    result = chain.loadProxyFromFile(self.proxyPath)
+    self.assertTrue(result['OK'], 'Cannot load proxy from file "%s": %s' % (self.proxyPath, result.get('Message')))
+
+    proxy = chain
+
+    print('Try to upload proxy with next params:\n Proxy: %s\nRFC: %s\nLivetime: %s\nUploadParams:\n%s\nProxyParams:\n%s\n' %
+          (proxy, rfcReq, timeReq, uploadParams, proxyParams))
+    result = self.proxyMgr.uploadProxy(proxy=proxy, restrictLifeTime=timeReq, rfcIfPossible=rfcReq)
+    print('\nRESULT: %s' % result)
```

Comment:
No assert(s) here?

Reply:
the tests have been fixed and implemented in Test_ProxyDB.py